Alexandria/boardgame-backend
# Feature/add boardgame

## Reference

[43](https://crema.atlassian.net/browse/BA-43)

## Overview

I created two new routes 

`GET  /search/addgame/:id `
`POST /search/addgame/:id`

The GET route will query BGG for all board games with the name the user searched for and return the results

The POST route will add the board game to the database after the user has selected a game. 

This is still a work in progress. The user can only add original games for now ( so no expansions) and they can only add games via the name of the board game and the name must be spelled correctly. 
These routes are not currently protected for testing purposes.  

### Acceptance Criteria

- [x] Be able to pull board game information from Boardgame Geek

- [x] Check if the board game is already in the local database before adding it

- [ ] Work out how to handle variations of boardgames

- [x] Work out the workflow to do all of this

### Demo

#### GET Route Test
![GET Route Test](https://cl.ly/22ec37d20374/Screen%252520Recording%2525202019-07-15%252520at%25252001.13%252520PM.gif)


#### POST Route Test
![POST Route Test](https://cl.ly/1222dcb453ec/Screen%252520Recording%2525202019-07-15%252520at%25252001.16%252520PM.gif)

#### User Workflow
<img width="994" alt="Screen Shot 2019-07-15 at 1 23 32 PM" src="https://user-images.githubusercontent.com/7613670/61239161-e168e800-a703-11e9-9f34-a930f5b53430.png">


## Reviewers' Checklist (go ahead, check 'em)


## Testing Instructions

Before testing this build you will need to have postgres installed and have a database already created.
Please look at the readme to see how to set up your environment variables.

1. Checkout branch feature/addBoardgame
2. If you have tables built in your database run `npx sequelize db:migrate:undo:all`, if not skip this step
3. To create all of the tables run: `npx sequelize db:migrate`
4. To seed data run: `npx sequelize db:seed:all`
5. Search board game route(GET) : http://localhost:YOUR_PORT/auth/search/addgame/:id
     - Set the `:id` param to number 1
     - The request body should be shaped like this:
`{"name":"BOARDGAME_NAME"}`
The above request will only give back a list of boardgames. The expected workflow is that the user will then select the game they are looking for and upon hitting submit the POST request will run. 

6. Add a new board game route (POST): http://localhost:YOUR_PORT/auth/search/addgame/:id
     - Set the `:id` param to number 1
     - To test to see if a duplicate will  be added, define your request body as below:
`{"bgGeekID":"14"}`
-To test adding  a new game to a collection define your request body as below:
`{"bgGeekID":"2223"}`
7. To view all of the boardgames that you have recently added you first need to login
     - Login route(POST):  http://localhost:YOURPORT/auth/login
     - Define your request body as below:
`{"email":"admin@gmail.com","password":"password"}`
     - Home route(GET):  http://localhost:YOUR_PORT/auth/home/:id
     - Set the `:id` param to number 1
